pkg/diagnostics/client/config_contexts.go

```diff
@@ -0,0 +1,289 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclientcmd "k8s.io/kubernetes/pkg/client/clientcmd"
+	kclientcmdapi "k8s.io/kubernetes/pkg/client/clientcmd/api"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+// ConfigContext diagnostics (one per context) validate that the client config context is complete and has connectivity to the master.
+type ConfigContext struct {
+	RawConfig   *kclientcmdapi.Config
+	ContextName string
+}
+
+const (
+	ConfigContextsName    = "ConfigContexts"
+	currentContextMissing = `Your client config specifies a current context of '{{.context}}'
+which is not defined; it is likely that a mistake was introduced while
+manually editing your config. If this is a simple typo, you may be
+able to fix it manually.
+The master creates a fresh client config when it is started; it may be
```

Comment:
the master doesn't create client configurations when started from a config file...